mmd.js

```diff
@@ -12,8 +12,8 @@
 			.replace(/!\[([^\]]*)]\(([^(]+)\)/g, '<img alt="$1" src="$2">')
 			.replace(/\[([^\]]+)]\(([^(]+)\)/g, '$1'.link('$2'))
 			.replace(/`([^`]+)`/g, '<code>$1</code>')
-			.replace(/\*\*([^*]+)\*\*/g, '<strong>$1</strong>')
-			.replace(/\*([^*]+)\*/g, '<em>$1</em>');
+			.replace(/(\*\*|__)(?=\S)([^\r]*?\S[*_]*)\1/g, '<strong>$2</strong>')
```

Comment:
Nicely done with the `\1`.
I don't fully get the purpose of the `?\S[*-]*`
There can not be any `\r`, they are removed by the `src.replace(/^\s+|\r|\s+$/g, '')....`